crashsimilarity/downloader.py

```diff
@@ -0,0 +1,76 @@
+from datetime import timedelta
+import logging
+
+from crashsimilarity import utils
+
+
+class Downloader(object):
+    _SUPER_SEARCH_URL = 'https://crash-stats.mozilla.com/api/SuperSearch'
+    _PROCESSED_CRASH_URL = 'https://crash-stats.mozilla.com/api/ProcessedCrash'
+    _BUGZILLA_BUG_URL = 'https://bugzilla.mozilla.org/rest/bug'
+
+    def __init__(self, cache=None):
+        self._cache = cache
+
+    def download_stack_traces_for_signature(self, signature, traces_num=100, period=timedelta(days=31)):
+        from_date = utils.utc_today() - period
+
+        key = ('traces_for_signature', signature, utils.utc_today(), from_date)
+        if self._cache and key in self._cache:
+            logging.debug('get data from cache')
+            return self._cache[key]
+
+        params = {
+            'signature': '=' + signature,
+            'date': ['>=' + str(from_date)],
+            '_facets': ['proto_signature'],
+            '_facets_size': traces_num,
+            '_results_number': 0
+        }
+        response = utils.get_with_retries(self._SUPER_SEARCH_URL, params)
+        records = self._json_or_raise(response)['facets']['proto_signature']
+        traces = set([r['term'] for r in records])
+
+        if self._cache:
+            self._cache[key] = traces
+        return traces
+
+    def download_crash_for_id(self, uuid):
```

Comment:
In this case, since the only way to get a specific crash from Socorro is via a UUID, the `for_id` is redundant.
It doesn't hurt to have a longer name though, so feel free to leave it as it is (when I say "Nit" it means that it's a minor thing and you can ignore it).